### PR TITLE
dev-python/icalendar: update to version 4.0.3

### PR DESCRIPTION
**Pull request by Cynerd**
Bug: https://bugs.gentoo.org/672530

Signed-off-by: Karel Kočí <cynerd@email.cz>

### CODE REVIEW

File: dev-python/icalendar/icalendar-4.0.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+PYTHON_COMPAT=( python2_7 python3_{5,6,7} pypy )
+
+inherit distutils-r1
+
+MY_PN="collective-${PN}"
```
Comment (mgorny) [inline dev-python/icalendar/icalendar-4.0.3.ebuild]: This isn't used anywhere.

File: dev-python/icalendar/icalendar-4.0.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+DEPEND="
+	${RDEPEND}
+	doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )
+	test? (	dev-python/python-dateutil:0[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/coverage[${PYTHON_USEDEP}] )"
+
+python_prepare_all() {
+	# reset conf.py to not read version from an installed instance
+	sed -e "s:pkg_resources.get_distribution('icalendar').version:'3.9.0':" \
```
Comment (mgorny) [inline dev-python/icalendar/icalendar-4.0.3.ebuild]: This is certainly wrong.

File: dev-python/icalendar/icalendar-4.0.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	dev-python/pytz[${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	dev-python/python-dateutil[${PYTHON_USEDEP}]"
+
+DEPEND="
+	${RDEPEND}
+	doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )
+	test? (	dev-python/python-dateutil:0[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/coverage[${PYTHON_USEDEP}] )"
```
Comment (mgorny) [inline dev-python/icalendar/icalendar-4.0.3.ebuild]: Kill the dep on coverage.

File: dev-python/icalendar/icalendar-4.0.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+SRC_URI="mirror://pypi/i/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~x86-fbsd"
+IUSE="doc test"
+DOCS="README.rst"
+
+RDEPEND="
+	dev-python/pytz[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/icalendar/icalendar-4.0.3.ebuild]: Sort.

File: dev-python/icalendar/icalendar-4.0.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+python_prepare_all() {
+	# reset conf.py to not read version from an installed instance
+	sed -e "s:pkg_resources.get_distribution('icalendar').version:'3.9.0':" \
+		-i docs/conf.py || die
+	distutils-r1_python_prepare_all
+}
+
+python_compile_all() {
+	if use doc; then
+		pushd docs > /dev/null
```
Comment (mgorny) [inline dev-python/icalendar/icalendar-4.0.3.ebuild]: `emake -C docs ...`

File: dev-python/icalendar/icalendar-4.0.3.ebuild (hunk 1 above)
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: You are right.

File: dev-python/icalendar/icalendar-4.0.3.ebuild (hunk 2 above)
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: Yep that is for sure. It is used only as a string in documentation I suspect but correct version should be used non the less. I replaced it with `${PV}` so no future bumps have to update this.

File: dev-python/icalendar/icalendar-4.0.3.ebuild (hunk 3 above)
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: Yep you are correct. It is not required. I also ran test and it looks like slotted dateutil is not required either so I dropped it.

File: dev-python/icalendar/icalendar-4.0.3.ebuild (hunk 4 above)
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: Should be now..

File: dev-python/icalendar/icalendar-4.0.3.ebuild (hunk 5 above)
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: Replaced

File: dev-python/icalendar/icalendar-4.0.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	# reset conf.py to not read version from an installed instance
+	sed -e "s:pkg_resources.get_distribution('icalendar').version:'${PV}':" \
+		-i docs/conf.py || die
+	distutils-r1_python_prepare_all
+}
+
+python_compile_all() {
+	if use doc; then
+		emake -C docs text
+		DOCS=( ${DOCS} docs/_build/text/*.txt )
```
[the remark this answers is not in the document]
Comment (sbraz) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: Don't we usually build HTML docs?

File: dev-python/icalendar/icalendar-4.0.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+DESCRIPTION="Package used for parsing and generating iCalendar files (RFC 2445)"
+HOMEPAGE="https://github.com/collective/icalendar"
+SRC_URI="mirror://pypi/i/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~x86-fbsd"
+IUSE="doc test"
+DOCS="README.rst"
```
[the remark this answers is not in the document]
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: Ok

File: dev-python/icalendar/icalendar-4.0.3.ebuild (hunk 6 above)
[the remark this answers is not in the document]
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: I have no idea what you are usually using. If it is html then be it html.

File: dev-python/icalendar/icalendar-4.0.3.ebuild [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+python_compile_all() {
+	if use doc; then
+		emake -C docs text
+		DOCS=( ${DOCS} docs/_build/text/*.txt )
+	fi
+}
+
+python_test() {
+	# From tox.ini
+	coverage run --source=src/icalendar --omit=*/tests/* --module \
```
[the remark this answers is not in the document]
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: Ups. I missed that. Ok that dependency is back in place and be it `pytest -vv`.

File: dev-python/icalendar/icalendar-4.0.3.ebuild (hunk 6 above)
[the remark this answers is not in the document]
Comment (sbraz) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: The function often looks like that:
```sh
python_compile_all() {
    if use doc; then
        sphinx-build docs docs/_build/html || die
        HTML_DOCS=( docs/_build/html/. )
    fi
}
```
I haven't really looked at this particular package but maybe you can do something similar.

[the remark this answers is not in the document]
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: I did not meant that I don't know how to do it. I already implemented it. Originally I used `python_install_all` and `local HTML_DOCS` but now I moved it back to `python_compile_all`.

File: dev-python/icalendar/icalendar-4.0.3.ebuild (hunk 8 above)
[the remark this answers is not in the document]
Comment (Cynerd) [reply dev-python/icalendar/icalendar-4.0.3.ebuild]: Ok? If you say so. I copied this part from previous version of package so I have no idea why `coverage` was previously used in tests but it is now dropped. I removed `coverage` use.

**Pull request closed without merging**